### PR TITLE
Tests for `network_utils`

### PR DESCRIPTION
**Pull request by strickvl**
Added tests for the `network_utils` code.

Wasn't fully sure of the best way to test this kind of network calls / scans, so happy to rework following any feedback that might come my way.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: tests/unit/utils/test_networking_utils.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    port = s.getsockname()[1]
+    s.close()
+    return port
+
+
+def test_port_available_works():
+    """Test if port_available works."""
+    open_port = find_open_port()
+    assert networking_utils.port_available(open_port)
+    assert isinstance(networking_utils.port_available(open_port), bool)
```
Comment (fa9r) [inline tests/unit/utils/test_networking_utils.py:36]:
```suggestion
    networking_utils.port_available(open_port) is True
```

File: tests/unit/utils/test_networking_utils.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+def test_port_is_open_on_local_host_works():
+    """Test if port_is_open_on_local_host works."""
+    open_port = find_open_port()
+    assert networking_utils.port_is_open("127.0.0.1", open_port) is not None
+    assert isinstance(
+        networking_utils.port_is_open("127.0.0.1", open_port), bool
+    )
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        sock.bind(("127.0.0.1", 8000))
+        assert networking_utils.port_is_open("127.0.0.1", 8000) is False
```
Comment (fa9r) [inline tests/unit/utils/test_networking_utils.py:70]: I think `sock.bind` could fail for 8000 if that port is already occupied on my local machine

```suggestion
        sock.bind(("127.0.0.1", open_port))
        assert networking_utils.port_is_open("127.0.0.1", open_port) is False
```

File: tests/unit/utils/test_networking_utils.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        assert available_port is None
+
+
+def test_port_is_open_on_local_host_works():
+    """Test if port_is_open_on_local_host works."""
+    open_port = find_open_port()
+    assert networking_utils.port_is_open("127.0.0.1", open_port) is not None
+    assert isinstance(
+        networking_utils.port_is_open("127.0.0.1", open_port), bool
+    )
```
Comment (fa9r) [inline tests/unit/utils/test_networking_utils.py:67]:
```suggestion
    assert networking_utils.port_is_open("127.0.0.1", open_port) is True
```

**Review by fa9r — APPROVED** [review]
Just a few minor suggestions, overall LGTM!

File: tests/unit/utils/test_networking_utils.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    s.listen(1)
+    port = s.getsockname()[1]
+    s.close()
+    return port
+
+
+def test_port_available_works():
+    """Test if port_available works."""
+    open_port = find_open_port()
+    networking_utils.port_available(open_port) is True
```
Comment (stefannica) [inline tests/unit/utils/test_networking_utils.py:35]:
```suggestion
    assert networking_utils.port_available(open_port) is True
```

... is probably what you meant.

**Review by stefannica — CHANGES_REQUESTED** [review]
One suggestion, looks great otherwise

**Review by stefannica — APPROVED** [review]

**Pull request merged**